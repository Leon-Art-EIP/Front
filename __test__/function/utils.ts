import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
