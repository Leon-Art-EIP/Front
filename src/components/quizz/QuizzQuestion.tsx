import { Badge } from "../lib/lib";

export interface QuizzQuestionProps {
  question: {
    question: string;
    answers: {
      text: string;
      selected: boolean;
    }[];
  };
  onSelectAnswer: (index: number) => void;
}

export default function QuizzQuestion(props: QuizzQuestionProps): JSX.Element {
  return (
    <div className="flex flex-col mt-24 gap-20">
      <span className="text-3xl font-semibold mx-32">{props.question.question}</span>
      <div className="flex justify-center select-none">
        <div className="flex flex-wrap justify-center w-1/2 gap-x-10 gap-y-5">
          {props.question.answers.map((answer, index) => (
            <span
              className={` px-6 py-4 rounded-[60px] cursor-pointer hover:border-[#cd5449] ${
                answer.selected ? "bg-[#FF7F74] text-white" : "bg-[#F4F4F4] text-black"
              }`}
              onClick={() => props.onSelectAnswer(index)}
              key={index}
            >
              <span className="text-lg">{answer.text}</span>
            </span>
          ))}
        </div>
      </div>
    </div>
  );
}
