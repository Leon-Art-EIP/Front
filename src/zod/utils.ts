export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
