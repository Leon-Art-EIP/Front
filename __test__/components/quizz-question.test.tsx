import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import QuizzQuestion from "../../src/components/quizz/QuizzQuestion";

test("QuizzQuestion snapshot", () => {
  testSnapshot(
    <QuizzQuestion
      location="alert"
      onSelectAnswer={() => {}}
      question={{
        question: "question",
        multipleChoice: false,
        answers: [{ text: "answer", selected: false }],
      }}
      setLocation={() => {}}
    />
  );
});
