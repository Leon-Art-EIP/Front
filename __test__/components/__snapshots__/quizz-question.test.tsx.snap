// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuizzQuestion snapshot 1`] = `
<div
  className="flex flex-col mt-5 gap-20"
>
  <span
    className="text-3xl font-semibold px-14 lg:px-32"
  >
    question
  </span>
  <div
    className="flex flex-col items-center gap-10 select-none"
  >
    <div
      className="flex flex-wrap flex-col lg:flex-row items-center justify-center w-1/2 gap-x-10 gap-y-5"
    >
      <span
        className="text-center px-6 py-4 rounded-[60px] cursor-pointer hover:bg-[#fca199] hover:text-white bg-[#F4F4F4] text-black"
        id="name-0"
        onClick={[Function]}
      >
        <span
          className="text-lg"
        >
          answer
        </span>
      </span>
    </div>
  </div>
</div>
`;
